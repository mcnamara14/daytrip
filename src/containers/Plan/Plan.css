.plan {
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.planContainer {
  padding-bottom: 65px;
  border-bottom: 1px solid #eee;
}

.plan h2 {
  color: #666;
  text-transform: uppercase;
  font-size: 1.4rem;
  margin-bottom: 10px;
}

.plan article {
  width: 21%;
  text-align: left;
}

.stars {
  float: left;
  margin: 0 10px 0 0;
  width: 100px;
  position: relative;
}

.redStars{
  border: 0;
  left: 0;
  overflow: hidden;
  position: absolute;
  top: 0;
 } 

 .stars img {
   height: auto;
   max-width: 100px;
 }

.reviewCount {
  color: #888;
}

.price {
  font-size: 1.2rem;
  color: #666;
}

.location {
  color: #888;
  font-size: 0.85rem;
  margin: 0 0 25px 0;
}

.stopImage {
  width: 100%;
  height: 190px;
  border-radius: 5px 5px 0 0;
  display: block;
  border-radius: 5px;
  background-size: cover;
  background-position: center;
  border: 1px solid #eee;
}

.stopInfo {
  height: 185px;
}

.routeImg {
  width: auto;
  margin-top: 16%;
}