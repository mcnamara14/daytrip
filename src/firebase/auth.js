import { auth } from './firebase';
import * as firebase from 'firebase';

// Email Sign Up
export const emailPasswordSignup = (email, password) =>
  auth.createUserWithEmailAndPassword(email, password);

  // Email Sign Up
export const googleSignup = () =>
auth.signInWithPopup(new firebase.auth.GoogleAuthProvider())

  // Email Sign In
export const emailPasswordSignin = (email, password) =>
auth.signInWithEmailAndPassword(email, password);

// Sign out
export const signOut = () =>
  auth.signOut();

  // Password Reset
export const passwordReset = (email) =>
  auth.sendPasswordResetEmail(email);

// Password Change
export const updatePassword = (password) =>
  auth.currentUser.updatePassword(password);
