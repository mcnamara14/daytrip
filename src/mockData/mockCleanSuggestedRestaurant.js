export const mockCleanSuggestedRestaurant = [
  {
    address: "1543 Champa St Denver, CO 80202",
    image_url: "https://s3-media2.fl.yelpcdn.com/bphoto/Twaes8jv0U3XOgGu6H9d-A/o.jpg",
    name: "Union Lodge No.1",
    price: "$$",
    rating: 5,
    review_count: 382
  }
];

export const mockDirtySuggestedRestaurant = [
  {
    image_url: "https://s3-media2.fl.yelpcdn.com/bphoto/Twaes8jv0U3XOgGu6H9d-A/o.jpg",
    name: "Union Lodge No.1",
    price: "$$",
    rating: 5,
    review_count: 382,
    location: {
      display_address : [
        "1543 Champa St",
        "Denver, CO 80202"
      ]
    }
  }
];