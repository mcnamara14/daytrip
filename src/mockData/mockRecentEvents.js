export const mockDirtyRecentEvents = {
  events: [
    {
      title: "T Swift",
      image: "https://s1.ticketm.allswifty.jpg",
      price: "$300",
      venue: "Boulder Theater",
      date: "2018-05-30 8:00 PM"
    }
  ]
};

