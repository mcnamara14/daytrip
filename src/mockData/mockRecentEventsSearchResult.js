export const mockCleanRecentEventsSearchResult = {
  "options": [
    {"id": "Zrt9dk32l", 
      "label": "T SwiftBoulder Theater2018-05-30 2018-05-27T17:13:38-06:00", 
      "value": 0}
  ]
}

export const mockCleanRecentEventsSearchResultNoStart = {
  "options": [
    {"id": "Zrt9dk32l", 
      "label": "T SwiftBoulder TheaterTBA", 
      "value": 0}
  ]
}
