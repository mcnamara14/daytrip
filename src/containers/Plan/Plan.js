import React, { Component } from 'react';
import { connect } from 'react-redux';
import './Plan.css';
import PropTypes from 'prop-types';
import * as firebase from 'firebase';
import 'firebase/database';
import { googleApiKey } from '../../apiCalls/apiKeys/googleApiKey';

export class Plan extends Component {
  constructor() {
    super();

    this.state = {
      selectedPlan: null
    }
  }

  componentDidMount() {
    let firebaseDb;

    if (this.props.user.userId !== null) {
      firebaseDb = firebase.database().ref().child('users').child(this.props.user.userId).child('selectedPlan');
    } else {
      firebaseDb = firebase.database().ref().child('users').child('anonymous').child('selectedPlan');
    };

    firebaseDb.once('value')
      .then((snapshot) => {
        return snapshot.val();
      }).then(selectedPlan => 
        this.setState({selectedPlan}))
  }

  render () {
    let city;
    let state;
    
    const { selectedPlan } = this.state;

    if (selectedPlan) {
      const {
        bar,
        restaurant,
        event
      } = this.state.selectedPlan;

      const {
        image: restaurantImage,
        title: restaurantTitle,
        location: restaurantLocation,
        rating: restaurantRating,
        reviews: restaurantReviews,
        type: restaurantType
      } = restaurant;

      const {
        image: barImage,
        title: barTitle,
        location: barLocation,
        rating: barRating,
        reviews: barReviews,
        type: barType
      } = bar;

      const {
        image: eventImage,
        title: eventTitle,
        location: eventLocation,
        price: eventPrice,
        reviews: eventDate
      } = event;

      if (this.props.location) {
        const { city, state } = this.props.location;
      } else {
        const { city, state } = this.props.user
      }
      

      const restaurantBgImage = {backgroundImage: "url(" + restaurantImage + ")"};
      const eventBgImage = {backgroundImage: "url(" + eventImage + ")"};
      const barBgImage = {backgroundImage: "url(" + barImage + ")"};

      const restaurantReview = restaurantRating * 20;
      const barReview = barRating * 20;
      const mapRestaurantLocation = restaurantLocation.replace(/\s+/g, '+').toLowerCase().replace(',', '');
      const mapEventLocation = eventLocation.replace(/\s+/g, '+').toLowerCase().replace(',', '');
      const mapBarLocation = barLocation.replace(/\s+/g, '+').toLowerCase().replace(',', '');

      const mapStyles = {
        width: '80%',
        height: '350px',
        border: '0',
        borderRadius: '5px'
      };

      return (
        <section className="planMain">
          <div className="planHeader"></div>
          <div className="planWrapper">
            <section className="planLeftContainer">
              <h1>Your plan</h1>
              <div className="planStop">
                <h2>Begin</h2>
                <div className="planImage" style={restaurantBgImage}></div>
                <div className="planInfo">
                  <h3>{restaurantTitle}</h3>
                  <div className="reviewContainer">
                    <div className="stars">
                      <div>
                        <img src={require('./assets/stars-gray.png')} />
                      </div>
                      <div className="redStars" style={{ "width": `${restaurantReview}%` }}>
                        <img src={require('./assets/stars-red.png')} />
                      </div>
                    </div>
                    <p className="reviewCount">{restaurantReviews} reviews</p>
                  </div>
                  <p className="location">{restaurantLocation}</p>
                </div>
              </div>
              <div className="planStop">
                <h2>Event</h2>
                <div className="planImage" style={eventBgImage}></div>
                <div className="planInfo">
                  <h3>{eventTitle}</h3>
                  <p className="location">{eventLocation}</p>
                  <p className="eventPrice">{eventPrice}</p>
                  <p className="eventDate">{eventDate}</p>
                </div>
              </div>
              <div className="planStop">
                <h2>After</h2>
                <div className="planImage" style={barBgImage}></div>
                <div className="planInfo">
                  <h3>{barTitle}</h3>
                  <div className="reviewContainer">
                    <div className="stars">
                      <div>
                        <img src={require('./assets/stars-gray.png')} />
                      </div>
                      <div className="redStars" style={{ "width": `${barReview}%` }}>
                        <img src={require('./assets/stars-red.png')} />
                      </div>
                    </div>
                    <p className="reviewCount">{barReviews} reviews</p>
                  </div>
                  <p className="location">{barLocation}</p>
                </div>
              </div>
            </section>
            <section className="planRightContainer">
              <h2>Directions:</h2>
              <h4>To the event</h4>
              <iframe src={`https://www.google.com/maps/embed/v1/directions?key=${googleApiKey}&origin=${mapRestaurantLocation}&destination=${mapEventLocation}+${city}+${state}&mode=walking`} style={mapStyles} ></iframe>
              <h4>From the event</h4>
              <iframe src={`https://www.google.com/maps/embed/v1/directions?key=${googleApiKey}&origin=${mapEventLocation}+${city}+${state}&destination=${mapBarLocation}&mode=walking`} style={mapStyles} ></iframe>
            </section>
          </div>
        </section>
      ) 
    } else {
      return <h1>Selected a plan</h1>
    }
  }
}

Plan.propTypes = {
  suggestedRestaurants: PropTypes.array
};

export const mapStateToProps = (state) => ({
  suggestedRestaurants: state.suggestedRestaurants,
  suggestedBars: state.suggestedBars,
  selectedEvent: state.selectedEvent,
  user: state.user,
  location: state.location
});

export default connect(mapStateToProps)(Plan);