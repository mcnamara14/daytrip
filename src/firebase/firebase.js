import * as firebase from 'firebase';
import { firebaseApiKey } from './apiKey';

const config = {
  apiKey: firebaseApiKey,
  authDomain: "daytrip-c5c3a.firebaseapp.com",
  databaseURL: "https://daytrip-c5c3a.firebaseio.com",
  projectId: "daytrip-c5c3a",
  storageBucket: "daytrip-c5c3a.appspot.com",
  messagingSenderId: "532630776936",
  };

if (!firebase.apps.length) {
  firebase.initializeApp(config);
}

const auth = firebase.auth();

export {
  auth
};