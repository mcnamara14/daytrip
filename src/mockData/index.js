import { mockDirtyRecentEvents } from './mockRecentEvents';
import { mockCleanRestaurantAndBarData } from './mockRestaurantAndBarData';

export {
  mockDirtyRecentEvents,
  mockCleanRestaurantAndBarData
}