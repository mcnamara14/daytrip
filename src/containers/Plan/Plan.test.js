import React from 'react';
import ReactDOM from 'react-dom';
import { Plan } from './Plan';
import { shallow } from 'enzyme';
import * as firebase from 'firebase';
import 'firebase/database';
import { mockCleanRestaurantAndBar, mockDirtyRecentEvents } from '../../mockData';

describe('Plan', () => {
  let wrapper;

  beforeEach(() => {
    const mockUser = {
      userId: '12345'
    }
    wrapper = shallow(<Plan user={mockUser} />, { disableLifecycleMethods: true });
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  // describe('render functionality', () => {
  //   let wrapper;

  //   beforeEach(() => {
  //     wrapper = shallow(<Plan user={mockUser} />, { disableLifecycleMethods: true });

  //     wrapper.setState({
  //       selectedPlan: {
  //         bar: {
  //           location: '555 Union Station, Denver, CO'
  //         },
  //         event: {
  //           location: '555 Union Station, Denver, CO'
  //         },
  //         restaurant: {
  //           location: '555 Union Station, Denver, CO'
  //         }
  //       }
  //     })
  //   });

  //   it('should force location to lowercase with no commas', () => {
  //   })
  })

