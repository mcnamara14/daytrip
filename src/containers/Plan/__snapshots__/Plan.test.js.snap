// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plan should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Plan
    user={
        Object {
            "userId": "12345",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Selected a plan",
    },
    "ref": null,
    "rendered": "Selected a plan",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Selected a plan",
      },
      "ref": null,
      "rendered": "Selected a plan",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
